apps/frontend/src/containers/LoginOptions.tsx

```diff
@@ -1,53 +1,156 @@
-import { clsx } from "clsx";
+import { useState } from "react";
+import { useApolloClient } from "@apollo/client";
+import { assertNever } from "@argos/util/assertNever";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { MailIcon } from "lucide-react";
+import { useForm } from "react-hook-form";
 import { useLocation } from "react-router-dom";
+import { z } from "zod";
 
-import { config } from "@/config";
 import { GitHubLoginButton } from "@/containers/GitHub";
 import { GitLabLoginButton } from "@/containers/GitLab";
 import { GoogleLoginButton } from "@/containers/Google";
-import { Link } from "@/ui/Link";
+import { graphql } from "@/gql";
+import { ButtonIcon } from "@/ui/Button";
+import { Form } from "@/ui/Form";
+import { FormRootToastError } from "@/ui/FormRootError";
+import { FormSubmit } from "@/ui/FormSubmit";
+import { FormTextInput } from "@/ui/FormTextInput";
+import { Separator } from "@/ui/Separator";
+
+import { AuthWithEmail } from "./AuthWithEmail";
+
+type Screen = "providers" | "verifyEmail";
 
 export function LoginOptions(props: {
   redirect?: string | null;
   isDisabled?: boolean;
   className?: string;
+  email: string;
+  onEmailChange: (email: string) => void;
 }) {
   const location = useLocation();
+  const { email, onEmailChange } = props;
   const redirect = props.redirect ?? location.pathname + location.search;
+  const [screen, setScreen] = useState<Screen>("providers");
+  switch (screen) {
+    case "providers": {
+      return (
+        <div className="flex w-full flex-col gap-6">
+          <EmailForm
+            defaultEmail={email}
+            onContinue={({ email }) => {
+              onEmailChange(email);
+              setScreen("verifyEmail");
+            }}
+          />
+          <Separator />
+          <div className="flex flex-col gap-4">
+            <GoogleLoginButton
+              redirect={redirect}
+              size="large"
+              className="w-full justify-center"
+              isDisabled={props.isDisabled}
+            >
+              Continue with Google
+            </GoogleLoginButton>
+            <GitHubLoginButton
+              redirect={redirect}
+              size="large"
+              className="w-full justify-center"
+              isDisabled={props.isDisabled}
+            >
+              Continue with GitHub
+            </GitHubLoginButton>
```

Comment:
Could be cool to have GitHub first

Reply:
It's less friction and less scary to use email or Google to signup. They will have to use GitHub later if they want to import a project. But at the end we will allow to create a project without GitHub. So better to keep this order I think.